wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenStatusProvider.java

```diff
@@ -901,7 +875,60 @@ protected void createPostRouting(List<ConnectRouting> connectRouting) {
           });
     }
 
-    if (!this.greenDnsRemoved) {
+    if (!this.greenTopologyChanged) {
+      // Green topology is not yet updated so different plugins may be misled and try to connect to green endpoint.
+      // Reroute green endpoints to IP addresses
+      this.roleByHost.entrySet().stream()
+          .filter(x -> x.getValue() == BlueGreenRole.TARGET)
+          .forEach(x -> {
+            final String greenHost = x.getKey();
+            final boolean isGreenHostInstance = rdsUtils.isRdsInstance(greenHost);
+            final String blueHost = rdsUtils.removeGreenInstancePrefix(greenHost);
+            BlueGreenInterimStatus interimStatus = this.interimStatuses[x.getValue().getValue()];
+
+            HostSpec blueHostSpec = interimStatus == null
+                ? this.hostSpecBuilder.host(blueHost).build()
+                : this.hostSpecBuilder.host(blueHost).port(interimStatus.port).build();
+            HostSpec greenHostSpec = interimStatus == null
+                ? this.hostSpecBuilder.host(greenHost).build()
+                : this.hostSpecBuilder.host(greenHost).port(interimStatus.port).build();
+
+            Optional<String> greenIp = this.hostIpAddresses.get(greenHost);
+            HostSpec greenHostSpecWithIp = greenIp == null || !greenIp.isPresent()
+                ? this.hostSpecBuilder.host(greenHost).build()
+                : (interimStatus == null
+                    ? this.hostSpecBuilder.host(greenIp.get()).build()
+                    : this.hostSpecBuilder.host(greenIp.get()).port(interimStatus.port).build());
+
+            // Check whether green host is already been connected with blue (no-prefixes) IAM host name.
+            List<HostSpec> iamHosts;
+            if (this.isAlreadySuccessfullyConnected(greenHost, greenHost)) {
```

Comment:
I don't understand this code to clearly so just wanted to double check that it makes sense to pass in greenHost twice here? The comment above mentions checking if green has been connected to blue so I expected one of these to be a blue host